$NetBSD: patch-ai,v 1.26 2010/02/25 13:01:23 martti Exp $

Make this pkgsrc friendly.
Add support for NetBSD 5.x and DragonFly BSD.

--- makedefs.orig	2010-02-03 22:58:58.000000000 +0200
+++ makedefs	2010-02-25 14:48:50.000000000 +0200
@@ -146,6 +146,8 @@
 		;;
   FreeBSD.8*)	SYSTYPE=FREEBSD8
 		;;
+ DragonFly.*)	SYSTYPE=DRAGONFLY
+		;;
   OpenBSD.2*)	SYSTYPE=OPENBSD2
 		;;
   OpenBSD.3*)	SYSTYPE=OPENBSD3
@@ -162,6 +164,8 @@
 		;;
    NetBSD.4*)	SYSTYPE=NETBSD4
 		;;
+   NetBSD.5*)	SYSTYPE=NETBSD5
+		;;
    BSD/OS.2*)	SYSTYPE=BSDI2
 		;;
    BSD/OS.3*)	SYSTYPE=BSDI3
@@ -214,13 +218,6 @@
 		esac
 		;;
    ULTRIX.4*)	SYSTYPE=ULTRIX4
-		if [ -f /usr/local/lib/libdb.a ]; then
-		    SYSLIBS="$SYSLIBS -ldb"
-		    CCARGS="$CCARGS -DHAS_DB"
-		    if [ -d /usr/local/include/db ]; then
-			CCARGS="$CCARGS -I/usr/local/include/db"
-		    fi
-		fi
 		for l in syslog resolv; do
 		    if [ -f /usr/local/lib/lib$l.a ]; then
 			SYSLIBS="$SYSLIBS -l$l"
@@ -252,25 +249,8 @@
 		esac;;
 		# Tested with RedHat 3.03 on 20020729.
     Linux.1*)	SYSTYPE=LINUX1
-		SYSLIBS="-ldb"
 		;;
     Linux.2*)	SYSTYPE=LINUX2
-		# Postfix no longer needs DB 1.85 compatibility
-		if [ -f /usr/include/db.h ]
-		then
-		    : we are all set
-		elif [ -f /usr/include/db/db.h ]
-		then
-		    CCARGS="$CCARGS -I/usr/include/db"
-		else
-		    # No, we're not going to try db1 db2 db3 etc.
-		    # On a properly installed system, Postfix builds
-		    # by including <db.h> and by linking with -ldb
-		    echo "No <db.h> include file found." 1>&2
-		    echo "Install the appropriate db*-devel package first." 1>&2
-		    echo "See the RELEASE_NOTES file for more information." 1>&2
-		    exit 1
-		fi
 		# GDBM locks the DBM .pag file after open. This breaks postmap.
 		# if [ -f /usr/include/gdbm-ndbm.h ]
 		# then
@@ -281,7 +261,6 @@
 		#     CCARGS="$CCARGS -DHAS_DBM -DPATH_NDBM_H='<gdbm/ndbm.h>'"
 		#     GDBM_LIBS=gdbm
 		# fi
-		SYSLIBS="-ldb"
 		for name in nsl resolv $GDBM_LIBS
 		do
 		    for lib in /usr/lib64 /lib64 /usr/lib /lib
@@ -383,25 +362,13 @@
 HP-UX.A.09.*)	SYSTYPE=HPUX9
 		SYSLIBS=-ldbm
 		CCARGS="$CCARGS -DMISSING_USLEEP"
-		if [ -f /usr/lib/libdb.a ]; then
-		    CCARGS="$CCARGS -DHAS_DB"
-		    SYSLIBS="$SYSLIBS -ldb"
-		fi
 		;;
 HP-UX.B.10.*)	SYSTYPE=HPUX10
 		CCARGS="$CCARGS `nm /usr/lib/libc.a 2>/dev/null |
 		    (grep usleep >/dev/null || echo '-DMISSING_USLEEP')`"
-		if [ -f /usr/lib/libdb.a ]; then
-		    CCARGS="$CCARGS -DHAS_DB"
-		    SYSLIBS=-ldb
-		fi
 		;;
 HP-UX.B.11.*)	SYSTYPE=HPUX11
 		SYSLIBS=-lnsl
-		if [ -f /usr/lib/libdb.a ]; then
-		    CCARGS="$CCARGS -DHAS_DB"
-		    SYSLIBS="$SYSLIBS -ldb"
-		fi
 		;;
 ReliantUNIX-?.5.43) SYSTYPE=ReliantUnix543
 		RANLIB=echo
